src/Fixer/ArrayNotation/AlphabeticalArrayKeySortFixer.php

```diff
@@ -0,0 +1,315 @@
+
+                    continue;
+                }
+
+                if ($clonedTokens[$i]->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_CLOSE)) {
+                    $i = $clonedTokens->findBlockStart(Tokens::BLOCK_TYPE_ARRAY_SQUARE_BRACE, $i);
+
+                    continue;
+                }
+
+                if ($clonedTokens[$i]->isGivenKind(T_ARRAY)) {
+                    $keyTokenIndex = $clonedTokens->getNextMeaningfulToken($clonedTokens->getNextMeaningfulToken($i));
+
+                    break;
+                }
+
+                if ($clonedTokens[$i]->equals(',') || $clonedTokens[$i]->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_OPEN)) {
+                    $keyTokenIndex = $clonedTokens->getNextMeaningfulToken($i);
+
+                    break;
+                }
+            }
+
+            $keyTokens = Tokens::fromArray(\array_slice($clonedTokens->toArray(), $keyTokenIndex, $startIndex - $keyTokenIndex - 1));
+
+            $key = implode('', array_map(static fn ($item) => $item->getContent(), $keyTokens->toArray()));
+        } else {
+            $keyTokenIndex = $clonedTokens->getPrevMeaningfulToken($startIndex);
+            $key = $clonedTokens[$keyTokenIndex]->getContent();
+        }
+
+        return [$key, $keyTokenIndex];
+    }
+
+    /**
+     * Sort nested array tokens and return the end index.
+     */
+    private function sortNestedTokens(Tokens $clonedTokens, int $index): int
+    {
+        [$nestedTokenStartIndex, $nestedTokenEndIndex] = $this->getArrayIndexes($clonedTokens, $index);
+
+        $nestedArrayTokens = Tokens::fromArray(\array_slice($clonedTokens->toArray(), $nestedTokenStartIndex, $nestedTokenEndIndex - $nestedTokenStartIndex + 1));
+
+        $this->sortTokens($nestedArrayTokens);
+
+        $clonedTokens->overrideRange($nestedTokenStartIndex, $nestedTokenEndIndex, $nestedArrayTokens);
+
+        if ($nestedArrayTokens->count() !== $clonedTokens->count()) {
+            $clonedTokens->clearEmptyTokens();
+        }
+
+        return $nestedTokenEndIndex;
+    }
+
+    /**
+     * Get start and end index of an array.
+     *
+     * @return int[]
+     */
+    private function getArrayIndexes(Tokens $tokens, int $startIndex): array
+    {
+        if ($tokens[$startIndex]->isGivenKind(T_ARRAY)) {
+            $startParentheses = $tokens->getNextTokenOfKind($startIndex, ['(']);
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $startParentheses);
+        } else {
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_ARRAY_SQUARE_BRACE, $startIndex);
+        }
+
+        return [
+            $startIndex,
+            $endIndex,
+        ];
+    }
+
+    /**
+     * Checks if the token represents a special key.
+     *
+     * @param int|string $value
+     */
+    private function isSpecialKey($value): bool
+    {
+        if (\is_int($value)) {
+            return false;
+        }
+
+        $tokens = Tokens::fromCode((string) $value);
```

Comment:
Let's avoid any type casting. We need to correct type? it means input type was not proper.